jsWebServer/src/waitForChildren.js

```diff
@@ -0,0 +1,25 @@
+const waitForChildren = (wrapperArray) => {
+  console.log("waiting for child processes to be ready...");
+  return new Promise((resolve, reject) => {
+    let counter = wrapperArray.length;
+    wrapperArray.forEach(({ id, child, busy }) => {
+      let readyData = "";
+      const onData = (data) => {
+        readyData += data;
+        if (readyData === "ready") {
+          console.log(`child ${id} is ready`);
+          counter -= 1;
+        }
+        if (counter === 0) {
+          wrapperArray.forEach(({ id, child, busy }) => {
+            child.stdout.removeListener("data", onData);
```

Comment:
onData is a local var of the forEach in line 5
that means each iteration a new onData is create with diffrent memory address
so when the counter will reset to zero
in line 15 you will remove the same onData from all childs, but only one of them is binded to it.
for example:
lets say we have two childerns
so to the  forEach loop in line 5 will create in its two iterations two onData
onData lets say in address 100 for child 0
onData lets say in address 200 for child 1
and lets say child 1 is the child that reset the counter to zero
then in line 15 
you will delete onData in address 200 from child 1 but also 0
which means the onData in address 100 will remain in child 0
(this is not big deal, i might be wrong unless the removeListeneder indeed working by address and not by the label "data").
anyway amazing work and great use of promises.

Reply:
What he said

Reply:
I think that you're right and even if not, writing the function your way is much cleaner in my opinion...
So I changed it.